### DISCUSSION TITLE
Config issue

### DISCUSSION THREAD

**Discussion by madey83** [idiap/coqui-ai-TTS#182] [Q&A] [answered]
import os

from trainer import Trainer, TrainerArgs

from TTS.tts.configs.shared_configs import BaseDatasetConfig
from TTS.tts.configs.vits_config import VitsConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.vits import Vits, VitsArgs, VitsAudioConfig
from TTS.tts.utils.speakers import SpeakerManager
from TTS.tts.utils.text.tokenizer import TTSTokenizer
from TTS.utils.audio import AudioProcessor

def main():
    
    # Path where you want to save the models outputs (configs, checkpoints and tensorboard logs)
	#output_path = os.path.dirname(os.path.abspath(__file__))
	output_path = "D:\\TTS-Training\\Vits-PL-Model-training\\Output"
	dataset_config = BaseDatasetConfig(
		formatter="ljspeech", meta_file_train="metadata.csv", 
        language="pl-pl",
        path="D:\\TTS-Training\\LJSpeech-1.1"
	)
	audio_config = VitsAudioConfig(
        # Training Sampling rate and the target sampling rate for resampling the downloaded dataset (Note: If you change this you might need to redownload the dataset !!)
        # Note: If you add new datasets, please make sure that the dataset sampling rate and this parameter are matching, otherwise resample your audios
		sample_rate=22050, 
        win_length=1024, 
        hop_length=256, 
        num_mels=80, 
        mel_fmin=0, 
        mel_fmax=None,
	) 
    vitsArgs = VitsArgs(
        use_language_embedding=True,
        embedded_language_dim=4,
        use_speaker_embedding=True,
        use_sdp=False,
    )

    config = VitsConfig(
        model_args=vitsArgs,
        audio=audio_config,
     .
     .
     .
)

i got this issue even if i copy available recipient..

train_vits_win_Thorsten-Voice.py", line 33
    vitsArgs = VitsArgs(
                        ^
IndentationError: unindent does not match any outer indentation level

Can i ask for help?

**Comment by eginhard** [comment, marked as answer]
Indentation in Python is important and you can't use it randomly or you'll get errors like this. You can use [black](https://github.com/psf/black) for automatic code formatting to help with this, it's in Coqui's development dependencies.

**Comment by madey83** [comment]
n/a
